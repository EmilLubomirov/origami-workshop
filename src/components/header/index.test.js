import React from "react";
import renderer from "react-test-renderer"
import getNavigation from "../../utils/navigation";
import TestingEnvironment from "../../test-utils/router";
import Header from "./index";

describe("Header Component", () =>{

    it('should return authLinks by getNavigation()', function () {

        const nav = getNavigation(true, 1234);

        const authNav = [
            {
                title: "Publications",
                path: "/"
            },
            {
                title: "Share Thoughts",
                path: "/share-thoughts"
            },

            {
                title: "Profile",
                path: `/profile/1234`
            }
        ];

        expect(nav).toStrictEqual(authNav);
    });

    it('should render authenticated routes', function () {

        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    id: 1234
                },
                isLoggedIn: true
            }
            }>
                <Header/>
            </TestingEnvironment>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render guest routes', function () {

        const tree = renderer.create(
            <TestingEnvironment value={{
                user: null,
                isLoggedIn: false
            }}>
              <Header/>
            </TestingEnvironment>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });
});