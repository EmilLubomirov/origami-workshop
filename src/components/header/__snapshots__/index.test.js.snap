// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt="bird-logo"
    className="logo"
    src="white-origami-bird.png"
  />
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </div>
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/share-thoughts"
      onClick={[Function]}
    >
      Share Thoughts
    </a>
  </div>
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/profile/1234"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>
</header>
`;

exports[`Header Component should render guest routes 1`] = `
<header
  className="navigation"
>
  <img
    alt="bird-logo"
    className="logo"
    src="white-origami-bird.png"
  />
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </div>
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
  <div
    className="header-link"
  >
    <a
      className="header-list-item"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</header>
`;
